tools/diffscripts/diffapi.ps1

```diff
@@ -0,0 +1,203 @@
+if ($AsmToolPath -ne $null) {
+    $asmtoolroot = $AsmToolPath
+}
+$asmtool = $asmtoolroot + "Microsoft.DotNet.AsmDiff.exe"
+
+
+$internalroot = Join-Path -Path $baseroot -Child "\iot-sdks-internals"
+
+if ($AsmToolPath -ne $null) {
+    $internalroot = $SDKInternalsPath
+}
+
+$compareDirectory = Join-Path -Path $internalroot -Child "\sdk_design_docs\CSharp\master"
+
+if ($Preview) {
+    $compareDirectory = Join-Path -Path $internalroot -Child "\sdk_design_docs\CSharp\preview"
+}
+
+# Hardcoded list of assembly names
+$filenames = @(
+    "Microsoft.Azure.Devices.Shared",
+    "Microsoft.Azure.Devices.Client",
+    "Microsoft.Azure.Devices",
+    "Microsoft.Azure.Devices.Provisioning.Client",
+    "Microsoft.Azure.Devices.Provisioning.Service",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Amqp",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Mqtt",
+    "Microsoft.Azure.Devices.Provisioning.Transport.Http",
+    "Microsoft.Azure.Devices.Provisioning.Security.Tpm"
+)
+
+# All of the files from the build
+$oldsetNames = @(
+    ((Join-Path -Path $reporoot -ChildPath "\shared\src\bin\Release\netstandard2.1\")+($filenames[0]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\iothub\device\src\bin\Release\netstandard2.1\")+($filenames[1]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\iothub\service\src\bin\Release\netstandard2.1\")+($filenames[2]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\device\src\bin\Release\netstandard2.1\")+($filenames[3]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\service\src\bin\Release\netstandard2.1\")+($filenames[4]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\transport\amqp\src\bin\Release\netstandard2.1\")+($filenames[5]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\transport\mqtt\src\bin\Release\netstandard2.1\")+($filenames[6]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\provisioning\transport\http\src\bin\Release\netstandard2.1\")+($filenames[7]+".dll")),
+    ((Join-Path -Path $reporoot -ChildPath "\security\tpm\src\bin\Release\netstandard2.1\")+($filenames[8]+".dll"))
+)
+
+# Create a list of the markdown files so we can compare them to the API doc directory
+$markdownoutput = @(
+    ($compareDirectory+ "\" + ($filenames[0]+".md")),
+    ($compareDirectory+ "\" + ($filenames[1]+".md")),
+    ($compareDirectory+ "\" + ($filenames[2]+".md")),
+    ($compareDirectory+ "\" + ($filenames[3]+".md")),
+    ($compareDirectory+ "\" + ($filenames[4]+".md")),
+    ($compareDirectory+ "\" + ($filenames[5]+".md")),
+    ($compareDirectory+ "\" + ($filenames[6]+".md")),
+    ($compareDirectory+ "\" + ($filenames[7]+".md")),
+    ($compareDirectory+ "\" + ($filenames[8]+".md"))
+)
+
+if ((Test-Path $compareDirectory) -ne $TRUE) {
+    Write-Host "You have not cloned the sdk internals repository. Or it is not in the same root repository collection location."
```

Comment:
...internals repository nor is it in the same...

Reply:
Better